ddtrace/contrib/internal/mongoengine/patch.py

```diff
@@ -1,5 +1,9 @@
+# TODO(mabdinur): Remove the pymongoengine integration, this integration does nothing special
+# it just uses the pymongo integration and creates unnecessary pin objects
```

Comment:
nit: Do we want to mark it as deprecated then? And fully remove in the next major version EOY?

Reply:
This is a weird one. We want to deprecate this integrations public api (patch, unpatch, get_versions). 

We will still provide support for this integration we just don't need the code in ddtrace.contrib.internal.mongoengine to do it. I am not sure how to communicate this. Customer's shouldn't be calling patch, unpatch, or get_versions anyway. 

Maybe log a deprecation warning in `ddtrace.patch_all()` when mongoengine=True.

Reply:
Yeah that makes sense. If we've already marked the integration as deprecated on the public side, then we can probably just deprecate the internal code along with it